miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -845,6 +866,10 @@ public ModelAndView editPropagateSamples(@PathVariable String sampleIds, ModelMa
           hasPlain = true;
         }
         LibraryDto library = new LibraryDto();
+        Library tempLibrary = new LibraryImpl();
+        tempLibrary.setSample(sample);
+        final String libraryAlias = namingScheme.generateLibraryAlias(tempLibrary);
```

Comment:
I agree with you @djcooke - this approach would be cleaner.
We do however have something that works, and works well for now and given that this task has already been dragging on for sometime I suggest we go with this code as it stands with a view to changing the approach in the future.
I am concerned with getting the develop branch deployed to our production server soon as I had hoped to do it around new year, since this issue isn't blocking / breaking anything I recommend we press on!